### DISCUSSION TITLE
Draw binary color chart on low memory mcu

### DISCUSSION THREAD

**Discussion by zhangzqs** [slint-ui/slint#5207] [Q&A]
I want to draw a chart on a low memory microcontroller using a software renderer. Since the software renderer doesn't support Path so I tried drawing the chart to a memory framebuffer and loading it via a slint Image component, but there wasn't enough memory. In fact, my chart is just a binary color, and there is no need to use RGB or RGBA8 buffer to occupy a lot of unnecessary memory resources. Is there any way to optimize?

**Comment by ogoffart** [comment]
We don't really have a solution for this at the moment.

What I was thinking would be to have some sort of equivalent of BorrowedOpenGLTextureBuilder but for the software renderer.
One could construct an Image from a `Box<dyn Fn(buffer: &mut[TargetPixel], line: usize, range: Range<usize>)>` or something like that. which the software renderer would call when it needs to draw it.  But that's not implemented at the moment.
